src/main/java/com/auction/auction/bid/service/BidService.java

```diff
@@ -21,17 +22,19 @@ public class BidService {
     @Autowired
     private KafkaProducerService kafkaProducerService;
 
+    @Override
     public Bid saveBid(String auctionId, Bid bid) {
         Auction auction = auctionService.getAuctionById(auctionId);
 
         auction.getBids().add(bid);
-        auctionService.saveOrUpdateAuction(auction);
+        auctionService.saveAuction(auction);
 
         kafkaProducerService.sendMessage("bid-notifications", "New bid placed on auction: " + auction.getTitle());
 
         return bidRepository.save(bid);
     }
 
+    @Override
     //place bid on auction
     public Bid placeBid(String auctionId, Bid bid) {
```

Comment:
this method is doing exactly the same as saveBid() method.

Reply:
Hi, it's one for placing bid and other for saving bid, for example when an user edits a bid.